staubli_val3_driver/val3/ros_server/decTrajPt.pgx

```diff
@@ -32,30 +32,34 @@
 
   // sequence
   l_nRetVal=fromBinary(rosTrajPtMsg.body.nData,4,"-4l",rosTrajPtMsg.jointTrajPt.nSeq)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // joints[10]
   for l_nIndex=0 to 9 step 1
     l_nRetVal=fromBinary(rosTrajPtMsg.body.nData[(l_nIndex*4)+4],4,"4.0l",rosTrajPtMsg.jointTrajPt.nJoints[l_nIndex])
-    if (l_nRetVal!=1)
+    if(l_nRetVal!=1)
       x_bFlag=false
       return
     endIf
   endFor
 
+  // Velocity buffer to smooth out last traj point
+  nVel[1] = nVel[0]
+  nVel[0] = rosTrajPtMsg.jointTrajPt.nVelocity
```

Comment:
Are these lines still supposed to be here?

Reply:
They need to be somewhere before line 53 of `decTrajPt.pgx`, so they could be moved somewhere else, but I'm not sure where would be better. I thought about moving them [here](https://github.com/ros-industrial/staubli_val3_driver/blob/1280ba7f587eb352b52955d0cd35976edc1269e8/staubli_val3_driver/val3/ros_server/decodeMsgBody.pgx#L31), but that does not really solve the issue of it happening in the decoder. I'm open to suggestions.

Reply:
What about adding that logic here:

https://github.com/ros-industrial/staubli_val3_driver/blob/67599be96dea874a3fd859b03cf5a452e163e53a/staubli_val3_driver/val3/ros_server/motionControl.pgx#L35-L43

my reasoning:

 - msg decoders should only be responsible for decoding messages, then pushing the results in the appropriate queues
 - the motion "thread" (ie: `motionControl`) is responsible for retrieving the messages from the queue, perhaps post-processing them, and finally pushing them into the controller's own queue using `movej(..)`

the "postprocessing" could be the velocity averaging you want to implement.

It requires a bit of state (ie: the two velocities you're using) and a bit of logic. To me this would seem like the proper place.

Reply:
Moved in 7a51706.